xla/service/gpu/llvm_gpu_backend/ptx_version_util.cc

```diff
@@ -33,23 +33,26 @@ DetermineHighestSupportedPtxVersionFromCudaVersion(
 
   // Mapping determined from
   // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#release-notes
-  // Examples:
-  // CUDA 11.0 -> PTX 7.0
-  // CUDA 11.1 -> PTX 7.1
-  // CUDA 12.0 -> PTX 8.0
-  // CUDA 12.4 -> PTX 8.4
-  // This versioning scheme is valid until CUDA 12.6
-  if (cuda_version < stream_executor::SemanticVersion{12, 6, 0}) {
-    return {cuda_version.major() - 4, cuda_version.minor(), 0};
+  switch (cuda_version.major() * 10 + cuda_version.minor()) {
+    case 110:
+      return {7, 0, 0};
+    case 111:
+      return {7, 1, 0};
+    case 120:
+      return {8, 0, 0};
+    case 124:
+      return {8, 4, 0};
+    case 126:
+      return {8, 5, 0};
+    case 128:
+      return {8, 7, 0};
+    case 129:
+      return {8, 8, 0};
+    case 130:
+      return {9, 0, 0};
+    default:
+      // Return maximum known PTX version.
```

Comment:
I think we still need a version check. As the unknown version may be CUDA 11.8 or any other older, unenumerated version. For those we should have an assertion, so such errors are caught at build time.

Also, the enumerated version list should probably be limited to the CUDA SDK versions supported by XLA. I'm not sure if  CUDA-11 is supported any more. Even if it is, it's likely limited to cuda-11.8, which introduced sm_90.